knowledge-base/adr/0020-generator-options.md

```diff
@@ -0,0 +1,60 @@
+| `--clearOutputDir`                                       | `--clearOutputDir`               | Delete all files in output directory                                                              | _same_                                                                                                                                                       |  false   |
+| `--generateJs`                                           | `--transpile`, `-t`              | Transpiles, default true                                                                          | Transpiles, default false. Can only be set if tsconfig is enabled. This should be explicitly stated in the documentation.                                    |  false   |
+| `--generatePackageJson`                                  | `--packageJson`                  | Writes a default package.json, default true.                                                      | Writes a default package.json, default true (needs core dependency). Optionally in the future: Writes a custom package.json if passed. Keep boolean for now. |   true   |
+| `--serviceMapping`                                       | `--serviceMapping`               | Considers a service mapping file. Shows a warning if none is given (differs from OData behavior). | Changes the service mapping path. Generates it always.                                                        |   false   |
+| `--tsConfig`                                             | `--tsConfig`                     | tsconfig.json file to overwrite the default "tsconfig.json".                                      | Writes a custom tsconfig.json if passed. Document that this should be used in combination with `transpile`.                                                  |  false   |
+| `--versionInPackageJson`                                 | `--packageVersion`               | Version in package.json, default is generator version.                                            | Version in package.json, default is `1.0.0`. Hide it.                                                                                                        |  false   |
+| **Options in OData, but not (yet in OpenAPI) generator** |
+| `--forceOverwrite`                                       | `--overwrite`                    | Overwrite files even it they exist, default false.                                                | _same_                                                                                                                                                       |  false   |
+| `--generateTypedocJson`                                  | `--typedocJson`                  | Writes a default typedoc.json, default true.                                                      | Remove/deprecate, use `files` instead.                                                                                                                       |    -     |
```

Comment:
I'm also fine with this. But as a user, I will now have to learn how to make a config file (which is really simple as a basic config).

Reply:
Agreed.